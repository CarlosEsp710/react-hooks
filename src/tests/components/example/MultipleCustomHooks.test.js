import React from "react";
import { shallow } from "enzyme";

import { MultipleCustomHooks } from "../../../components/examples/MultipleCustomHooks";
import { useCounter } from "../../../hooks/useCounter";
import { useFetch } from "../../../hooks/useFetch";

jest.mock("../../../hooks/useFetch");
jest.mock("../../../hooks/useCounter");

describe("test <MultipleCustomHooks />", () => {
  beforeEach(() =>
    useCounter.mockReturnValue({
      counter: 10,
      increment: () => {},
    })
  );

  test("should return component <MultipleCustomHooks />", () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });

    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should return data when done loading", () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: "Carlos",
          quote: "Hello, world!",
        },
      ],
      loading: false,
      error: null,
    });

    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper.find(".alert").exists()).toBe(false);
    expect(wrapper.find(".mb-3").text().trim()).toBe("Hello, world!");
    expect(wrapper.find("footer").text().trim()).toBe("Carlos");
  });
});
