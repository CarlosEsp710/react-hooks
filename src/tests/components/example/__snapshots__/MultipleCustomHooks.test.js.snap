// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <MultipleCustomHooks /> should return component <MultipleCustomHooks /> 1`] = `
<Fragment>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
</Fragment>
`;
